File: DSA/LeetCode/Hashmaps/506-Relative-Ranks.java
class Solution {
    public String[] findRelativeRanks(int[] scores) {

        Integer[] sortedScores = Arrays.stream(scores).boxed().sorted((a, b) -> b - a).toArray(Integer[]::new);
        Map<Integer, Integer> placesMap = new HashMap<>();
        for (int i = 0; i < sortedScores.length; i++) {
            placesMap.put(sortedScores[i], i + 1);
        }
        Map<Integer, String> places = new HashMap<>();
        places.put(1, "Gold Medal");
        places.put(2, "Silver Medal");
        places.put(3, "Bronze Medal");

        String[] res = new String[scores.length];
        for (int i = 0; i < scores.length; i++) {
            int score = scores[i];
            int place = placesMap.get(score);
            res[i] = places.getOrDefault(place, Integer.toString(place));
        }
        return res;
    }
}

File: DSA/LeetCode/Hashmaps/506-Relative-Ranks.py
class Solution(object):
    def findRelativeRanks(self, scores):
        """
        :type score: List[int]
        :rtype: List[str]
        """
        sorted_scores = sorted(scores, reverse=True)
        places_map = {}
        for i in range(len(sorted_scores)):
            places_map[sorted_scores[i]] = i + 1

        places = {
            1: "Gold Medal",
            2: "Silver Medal",
            3: "Bronze Medal",
        }

        res = []
        for score in scores:
            place = places_map[score]
            res.append(places.get(place, str(place)))

        return res
        